src/Land/Model/SoilWaterParameterizations.jl

```diff
@@ -0,0 +1,512 @@
+
+"""
+AbstractImpedanceFactor{FT <: AbstractFloat}
+
+"""
+abstract type AbstractImpedanceFactor{FT <: AbstractFloat} end
+
+"""
+AbstractViscosityFactor{FT <: AbstractFloat}
+"""
+abstract type AbstractViscosityFactor{FT <: AbstractFloat} end
+
+"""
+AbstractMoistureFactor{FT <:AbstractFloat}
+"""
+abstract type AbstractMoistureFactor{FT <: AbstractFloat} end
+
+
+"""
+AbstractsHydraulicsModel{FT <: AbstractFloat}
+
+    Hydraulics model is used in the moisture factor in hydraulic conductivity and in the matric potential. The single hydraulics model choice sets both of these.
+"""
+abstract type AbstractHydraulicsModel{FT <: AbstractFloat} end
+
+
+"""
+    vanGenuchten{FT} <: AbstractHydraulicsModel{FT}
+
+The necessary parameters for the van Genuchten hydraulic model; defaults are for Yolo light clay.
+# Fields
+
+$(DocStringExtensions.FIELDS)
+"""
+struct vanGenuchten{FT} <: AbstractHydraulicsModel{FT}
+    "Exponent parameter - using in matric potential"
+    n::FT
+    "used in matric potential. The inverse of this carries units in the expression for matric potential (specify in inverse meters)."
+    α::FT
+    "Exponent parameter - determined by n, used in hydraulic conductivity"
+    m::FT
+    function vanGenuchten{FT}(; n::FT = FT(1.43), α::FT = FT(2.6)) where {FT}
+        new(n, α, 1 - 1 / FT(n))
+    end
+end
+
+"""
+    BrooksCorey{FT} <: AbstractHydraulicsModel{FT}
+
+The necessary parameters for the Brooks and Corey hydraulic model.
+
+Defaults are chosen to somewhat mirror the Havercamp/vG Yolo light clay hydraulic conductivity/matric potential.
+# Fields
+
+$(DocStringExtensions.FIELDS)
+"""
+Base.@kwdef struct BrooksCorey{FT} <: AbstractHydraulicsModel{FT}
+    "ψ_b - used in matric potential. Units of meters."
+    ψb::FT = FT(0.1656)
+    "Exponent used in matric potential and hydraulic conductivity."
+    m::FT = FT(0.5)
+end
+
+"""
+    Haverkamp{FT} <: AbstractHydraulicsModel{FT}
+
+The necessary parameters for the Haverkamp hydraulic model for Yolo light clay.
+
+Note that this only is used in creating a hydraulic conductivity function, and another formulation for matric potential must be used.
+# Fields
+$(DocStringExtensions.FIELDS)
+"""
+struct Haverkamp{FT} <: AbstractHydraulicsModel{FT}
+    "exponent in conductivity"
+    k::FT
+    "constant A (units of cm^k) using in conductivity. Our sim is in meters"
+    A::FT
+    "Exponent parameter - using in matric potential"
+    n::FT
+    "used in matric potential. The inverse of this carries units in the expression for matric potential (specify in inverse meters)."
+    α::FT
+    "Exponent parameter - determined by n, used in hydraulic conductivity"
+    m::FT
+    function Haverkamp{FT}(;
+        k::FT = FT(1.77),
+        A::FT = FT(124.6 / 100.0^1.77),
+        n::FT = FT(1.43),
+        α::FT = FT(2.6),
+    ) where {FT}
+        new(k, A, n, α, 1 - 1 / FT(n))
+    end
+end
+
+"""
+struct MoistureIndependent{FT} <: AbstractMoistureFactor{FT} end
```

Comment:
```suggestion
    MoistureIndependent{FT} <: AbstractMoistureFactor{FT}
```

Reply:
ok! this is just a more compact way of writing that?

Reply:
oh wait, nevermind. this is in the documentation. sorry!